src/main/java/org/apache/sling/security/impl/ReferrerFilter.java

```diff
@@ -260,16 +264,32 @@ private Pattern[] createRegexPatterns(final String[] regexps) {
         return patterns.toArray(new Pattern[0]);
     }
 
+    private String[] mergeValues(String[] primary, List<ReferrerFilterAmendment> amendments,
```

Comment:
why not returning a collection here? IIUC it is later on anyways converted to a list

Reply:
Honestly, minimizing changes, but collections are fine or needed 4/5 times so it makes sense to change.